sdk/src/azure/iot/az_iot_adu_client.c

```diff
@@ -804,6 +807,28 @@ AZ_NODISCARD az_result az_iot_adu_client_parse_update_manifest(
               _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
             }
           }
+          /*
+          * C SDK will not support delta updates at this time, so relatedFiles,
+          * downloadHandler, and mimeType are not exposed or processed.
+          */
+          else if (az_json_token_is_text_equal(
+                  &ref_json_reader->token,
+                  AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_RELATED_FILES)))
+          {
+            _az_RETURN_IF_FAILED(az_json_reader_skip_children(ref_json_reader));
+          }
+          else if (az_json_token_is_text_equal(
+                  &ref_json_reader->token,
+                  AZ_SPAN_FROM_STR(AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_DOWNLOAD_HANDLER)))
+          {
+            _az_RETURN_IF_FAILED(az_json_reader_skip_children(ref_json_reader));
```

Comment:
I think these should all have 
```suggestion
            _az_RETURN_IF_FAILED(az_json_reader_skip_children(ref_json_reader));
            _az_RETURN_IF_FAILED(az_json_reader_next_token(ref_json_reader));
```

The skip children is a protection against the possibility of there being an array. It's a no-op if there isn't an array.

Reply:
Having both causing parsing issues when it is an object - I'll take a look at what happens if it's not an object though

Reply:
The following scenarios work with just `skip_children`, so it looks like it parses through fine since we want to ignore the contents anyways! There's a `az_json_reader_next_token` at the end of the switch statement that processes the manifest which either moves past the contents of a non-object or the closing bracket of an object.

`"relatedFiles": null` 
`"relatedFiles": "testString"`

Reply:
Ahhh I see. Checked on the skip children API and if it's a property name then it advances to the next token before skipping object/arrays.